From 926e1fc099b07793d76567642719892c3df83bda Mon Sep 17 00:00:00 2001
From: Timo Herbrecher <t.herbrecher@gateware.de>
Date: Tue, 9 Jul 2019 15:36:28 +0200
Subject: [PATCH 2/7] tqmls1043a: Add initial board support

Add initial board support for TQMLS1043A processing module based on NXP
LS1043A CPU on MBLS10xxA baseboard.

Signed-off-by: Timo Herbrecher <t.herbrecher@gateware.de>
---
 arch/arm64/boot/dts/freescale/Makefile        |  1 +
 .../freescale/fsl-tqmls1043a-mbls10xxa.dts    | 44 ++++++++++++++
 .../boot/dts/freescale/fsl-tqmls1043a.dtsi    | 60 +++++++++++++++++++
 3 files changed, 105 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-tqmls1043a-mbls10xxa.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-tqmls1043a.dtsi

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 86b81b8fb013..b870317d2d3f 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -29,4 +29,5 @@ dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2088a-qds.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2088a-rdb.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-lx2160a-qds.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-lx2160a-rdb.dtb
+dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-tqmls1043a-mbls10xxa.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-tqmls1046a-mbls10xxa.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-tqmls1043a-mbls10xxa.dts b/arch/arm64/boot/dts/freescale/fsl-tqmls1043a-mbls10xxa.dts
new file mode 100644
index 000000000000..7ab8e184dd74
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-tqmls1043a-mbls10xxa.dts
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Device Tree file for TQMLS1043A SoM on MBLS10xxA from TQ
+ *
+ * Copyright 2019 TQ-Systems GmbH
+ */
+
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+
+#include "fsl-tqmls1043a.dtsi"
+#include "fsl-tqmls10xxa-mbls10xxa.dtsi"
+
+/ {
+	model = "TQ TQMLS1043A SoM on MBLS10xxA board";
+	compatible = "tqc,tqmls1043a", "fsl,ls1043a";
+
+	aliases {
+		serial0 = &duart0;
+		serial1 = &duart1;
+		qsgmii_s1_p1 = &qsgmii1_phy1;
+		qsgmii_s1_p2 = &qsgmii1_phy2;
+		qsgmii_s1_p3 = &qsgmii1_phy3;
+		qsgmii_s1_p4 = &qsgmii1_phy4;
+		qsgmii_s2_p1 = &qsgmii2_phy1;
+		qsgmii_s2_p2 = &qsgmii2_phy2;
+		qsgmii_s2_p3 = &qsgmii2_phy3;
+		qsgmii_s2_p4 = &qsgmii2_phy4;
+	};
+
+	chosen {
+		stdout-path = "serial1:115200n8";
+	};
+};
+
+&esdhc {
+	cd-gpios = <&gpio3 2 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&gpio3 3 GPIO_ACTIVE_HIGH>;
+};
+
+#include "fsl-ls1043-post.dtsi"
+#include "fsl-tqmls10xxa-mbls10xxa-fman.dtsi"
+
diff --git a/arch/arm64/boot/dts/freescale/fsl-tqmls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-tqmls1043a.dtsi
new file mode 100644
index 000000000000..c6ba76a9dc2c
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-tqmls1043a.dtsi
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Device Tree Include file for LS1043A based SoM of TQ
+ *
+ * Copyright 2019 TQ-Systems GmbH
+ */
+
+#include "fsl-ls1043a.dtsi"
+#include "fsl-tqmls10xxa.dtsi"
+
+&qspi {
+	num-cs = <2>;
+	bus-num = <0>;
+	status = "okay";
+	fsl,qspi-has-second-chip;
+
+	qflash0: mx66u51235f@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <62500000>;
+		reg = <0>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot@0 {
+				label = "U-Boot-PBL";
+				reg = <0x0 0xd0000>;
+			};
+
+			env@d0000 {
+				label = "U-Boot Environment";
+				reg = <0xd0000 0x10000>;
+			};
+
+			fman@e0000 {
+				label = "Fman Microcode";
+				reg = <0xe0000 0x10000>;
+			};
+
+			dtb@f0000 {
+				label = "DTB";
+				reg = <0xf0000 0x10000>;
+			};
+
+			linux@100000 {
+				label = "Linux";
+				reg = <0x100000 0x1700000>;
+			};
+
+			rootfs@1800000 {
+				label = "RootFS";
+				reg = <0x1800000 0x2800000>;
+			};
+		};
+	};
+};
-- 
2.17.1

